### PR TITLE
NIFI-3585: Add DatabaseAdapter impl for MS SQL 2008

### PR DESCRIPTION
**Pull request by mattyb149**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by mattyb149** [comment]
@patricker do you mind taking a look at this? I am no MS SQL expert by any means, but I tested GenerateTableFetch and QueryDatabaseTable and they seem ok to me.

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/db/impl/MSSQL2008DatabaseAdapter.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        }
+
+        if (StringUtils.isEmpty(columnNames) || columnNames.trim().equals("*")) {
+            query.append("*");
+        } else {
+            query.append(columnNames);
+        }
+
+        if (limit != null && offset != null) {
+            query.append(", ROW_NUMBER() OVER(ORDER BY ID asc) rnum");
```
Comment (jfrazee) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/db/impl/MSSQL2008DatabaseAdapter.java]: @mattyb149 I don't think there's a guarantee that the `ORDBER BY` column is `ID`. You meant for this to be `orderByClause`, right?

**Comment by jfrazee** [comment]
@mattyb149 I think I'll be able to say LGTM if this `orderByClause` thing gets fixed. Right now I think it only works on tables where the max value column is ID.

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/db/impl/MSSQL2008DatabaseAdapter.java (hunk 1 above)
Comment (mattyb149) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/db/impl/MSSQL2008DatabaseAdapter.java]: Yeah I think you're right, will change and re-test, good catch!

**Comment by jfrazee** [comment]
@mattyb149 Ok LGTM now so +1

I tested with [NIFI-3585.xml](https://gist.github.com/jfrazee/63fc76f25bc5cdcfaf66d246cf232d19) on an Amazon RDS SQL Server instance using the Orders table + OrderID column in the Northwind database [1,2,3].

1. https://northwinddatabase.codeplex.com
2. http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/SQLServer.Procedural.Importing.html#SQLServer.Procedural.Importing.Native.Using.Restore
3. https://aws.amazon.com/blogs/database/migrating-microsoft-sql-server-enterprise-workloads-to-amazon-rds-part-1/

**Pull request closed without merging**